Failure in `test_timeline_status` (timeline start LSN varies across restart)

This failure was seen on a PR that only changed some pageserver http code.

The test appears to be asserting that after a safekeeper restart, the start LSN for a timeline is unchanged, but it has changed.

https://neon-github-public-dev.s3.amazonaws.com/reports/pr-5439/6383580706/index.html#suites/82004ab4e3720b47bf78f312dabe7c55/8b7b3aa1563560d1


```
AssertionError: assert Lsn("0/14EEA78") == Lsn("0/0")
 +  where Lsn("0/14EEA78") = SafekeeperTimelineStatus(acceptor_epoch=4, pg_version=160000, flush_lsn=Lsn("0/1507F10"), commit_lsn=Lsn("0/1507F10"), timeline_start_lsn=Lsn("0/14EEA78"), backup_lsn=Lsn("0/14EEA78"), peer_horizon_lsn=Lsn("0/1507F10"), remote_consistent_lsn=Lsn("0/14EEA78")).timeline_start_lsn
test_runner/regress/test_wal_acceptor.py:831: in test_timeline_status
    assert tli_status.timeline_start_lsn == timeline_start_lsn
E   assert Lsn("0/14EEA78") == Lsn("0/0")
E    +  where Lsn("0/14EEA78") = SafekeeperTimelineStatus(acceptor_epoch=4, pg_version=160000, flush_lsn=Lsn("0/1507F10"), commit_lsn=Lsn("0/1507F10"), timeline_start_lsn=Lsn("0/14EEA78"), backup_lsn=Lsn("0/14EEA78"), peer_horizon_lsn=Lsn("0/1507F10"), remote_consistent_lsn=Lsn("0/14EEA78")).timeline_start_lsn
```

## Steps to reproduce


## Expected result


## Actual result


## Environment


## Logs, links
-

@jcsp 
I read the test and have found out why it may have failed
The test does the following
1. create an endpoint and reads the timeline status from 1st safekeeper
2. stop and start the same endpoint and read timeline status again from the 1st safekeeper
3. Assert `timeline_start_lsn` is same in both responses
```
test_runner/regress/test_wal_acceptor.py:831: in test_timeline_status
    assert tli_status.timeline_start_lsn == timeline_start_lsn
E   assert Lsn("0/14EEA78") == Lsn("0/0")
```
Safekeeper creates timeline state locally with `timeline_start_lsn` as `Lsn(0)` on receving the `Greeting` message
https://github.com/neondatabase/neon/blob/78dde318270fc4c7a3e226af6d1c4ecf6731a61d/safekeeper/src/safekeeper.rs#L249

`timeline_start_lsn` is only updated after handling the `Elected` message
https://github.com/neondatabase/neon/blob/78dde318270fc4c7a3e226af6d1c4ecf6731a61d/safekeeper/src/safekeeper.rs#L780-L785

The assertion failure shows the earlier `timeline_start_lsn` as `Lsn::INVALID` so it may have handled the first status api call before processing the `Elected` message

--- a/test_runner/regress/test_wal_acceptor.py
+++ b/test_runner/regress/test_wal_acceptor.py
@@ -801,6 +801,9 @@ def test_timeline_status(neon_env_builder: NeonEnvBuilder, auth_enabled: bool):
         wa_http_cli_debug = wa.http_client(auth_token=env.auth_keys.generate_safekeeper_token())
         wa_http_cli_debug.check_status()
 
+    # create a dummy table to wait for timeline initialization in safekeeper
+    endpoint.safe_psql("create table wait_for_sk()")
+
     # fetch something sensible from status
     tli_status = wa_http_cli.timeline_status(tenant_id, timeline_id)
     epoch = tli_status.acceptor_epoch

